src/sage/schemes/hyperelliptic_curves/jacobian_endomorphisms.py

```diff
@@ -0,0 +1,465 @@
+# -*- coding: utf-8 -*-
+r"""
+Endomorphism rings of Jacobians of hyperelliptic curves.
+
+Given the Jacobian `J` of a hyperelliptic curve over a number field `K`,
+the class ``EndomorphismRing`` constructs the ring `\textup{End}_{K}(J)`
+of endomorphisms of J which are defined over `K`.
+
+Currently the only functionality is for genus 2 curves over the rational
+numbers `\QQ`, for which there are two methods associated to an instance
+`E` of ``EndomorphismRing``:
+
+- ``is_absolutely_field(E)``
+- ``is_absolutely_trivial(E)``
+
+``is_absolutely_field(E)`` determines whether the absolute endomorphism
```

Comment:
All throughout, I think we should use "geometric" instead of "absolute".  
For example, we say "geometrically smooth", and not "absolutely smooth".

Reply:
Done.